### ISSUE TITLE
Extra allocation in quad_tris

### ISSUE THREAD

**Issue by lnicola** [bzar/sloper#1]
You can avoid some allocations by using the [`Iterator::chain`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.chain) method in `quad_tris` instead of allocating a `Vec`.

**Comment by bzar** [comment]
Hey, thanks for the pointer! This was a quick couple hour doodle and I didn't get quite that far, since printing stuff made with this quickly took priority. I'll keep this issue open if you don't mind so I'll remember it when I work on the code again.

**Comment by lnicola** [comment]
Yeah, don't worry, I just saw your "what are you doing this week" post and took a quick look. Of course, you already knew about `Iterator::chain` since you used it in `main`.
